### ISSUE TITLE
[Bug] Tokenizer won't open V2 Applications

### ISSUE THREAD

**Issue by BokuNC** [MrPrimate/tokenizer#182]
Foundry V12.331
System: World of Darkness 5e v5.0.7 (Supposedly using Application v2)
------

None of the Tokenizer functions appear to be working, clicking of the actor picture does nothing (even with shift held down), the title bar option doesn't show up when activated. I have all permissions needed for it to work, but it doesn't.

**Comment by MrPrimate** [comment]
For the avatar click most (all?) other systems use the Foundry default method of having an img property of `data-edit` with either `img` or `prototypeToken.texture.src` set to identify the target. Other modules that rely on this core listener/property will also likely not work without adjustments.

The titlebar option is not currently working with v2 sheets.

**Comment by BokuNC** [comment]
Yeah, i don't really understand why it stopped working at all. I did request that they make it compatible again, but all that i got was that it probably stopped because of the "application v2" and that they couldn't really do much to solve it(?). So my hand are tied, since i have pretty much zero coding knowledge.

**Comment by Veilza** [comment]
My initial assumption was that since it was reported as completely broken it was just another module that hadn't been updated yet; that was actually my bad for not doing my own investigation first and now I've done some looking into this on my end to give better insight.

I will say this seems at least partly AppV2-related. But I do want to clear up the issue from @BokuNC that "clicking the actor picture does nothing" actually isn't an issue; I tested just now and Tokenizer is 100% functioning as expected within the system. So I don't know if it's a different module or anything causing that, but at the very least I can say the system's sheet isn't interfering with Tokenizer's.

The title bar option definitely isn't working though and I tested that with Into The Odd as well, one of the other systems I've found to use AppV2, so I think that actually is AppV2 related.

It's working on D&D5E because they're not **actually** using AppV2 yet. D&D5E's system creates a class called "ActorSheetV2" which is referring to V2 of the *system sheet*, not the actual ApplicationV2 framework as core Foundry calls it.

**Comment by Veilza** [comment]
Specifically for the header button not appearing: AppV2 defines header controls in _getHeaderControls() now and it doesn't seem like it'll re-add the getActorSheetHeaderButtons hook until V13, as per this issue over here: https://github.com/foundryvtt/foundryvtt/issues/11668

So honestly that functionality might remain broken until V13 when that hook is re-added, since that seems like what Tokenizer uses to add its header control.

**Comment by BokuNC** [comment]
Veilza seems to have found something on the system that was causing a problem and its supposed to be fixed for the next update. This does seem to close this "issue".

Thank you all for the attention and patience!

**Comment by Fabrice-Schoeneberger** [comment]
Also have this issue on V13.342 with DnD5e V5.0.2. Clicking the picture on a character does not open a tokenizer window

**Comment by MrPrimate** [comment]
> Also have this issue on V13.342 with DnD5e V5.0.2. Clicking the picture on a character does not open a tokenizer window

Is the sheet in edit mode?

**Comment by MrPrimate** [comment]
> Yes, the sheet is in edit mode. I can open the tokenizer window via the right-click option on a token in the actors tab. But not via character sheet, whether in edit mode or not

Well, what version of Tokenizer are you using? Have you tried with other modules disabled? (And are you using the core character sheet?) And are there any errors in the console logwhen opening a sheet?

**Comment by Fabrice-Schoeneberger** [comment]
- Tokenizer version 4.4.4
- I have tried it with all other modules disabled but 1 (The Ripper93's module hub, with I used for debugging)
- I'm in the newest version DnD5e V5.0.2 which uses it's own character sheet system I think. But if that is what you mean then yes
- I get one error when trying to open tokenizer this way:
```
Uncaught TypeError: el.parentElement is null
    _updatePosition foundry.mjs:27388
    setPosition foundry.mjs:27364
    _onSearchFilter foundry.mjs:29810
    timeoutId foundry.mjs:692
    setTimeout handler*debounce/< foundry.mjs:691
    filter foundry.mjs:28660
    bind foundry.mjs:28635
    _onRender foundry.mjs:29527
    _doEvent foundry.mjs:27654
    #render foundry.mjs:27000
    #try foundry.mjs:7130
    add foundry.mjs:7103
    add foundry.mjs:7101
    render foundry.mjs:26925
    render foundry.mjs:29407
    browse foundry.mjs:29397
    linkDefaultSheets hooks.js:342
    jQuery 8
    linkDefaultSheets hooks.js:320
    jQuery 2
    linkDefaultSheets hooks.js:315
    #call foundry.mjs:23610
    callAll foundry.mjs:23569
    #callHooks foundry.mjs:27700
    #dispatchEvent foundry.mjs:27684
    _doEvent foundry.mjs:27657
    promise callback*_doEvent foundry.mjs:27655
foundry.mjs:27388:76
```

**Comment by MrPrimate** [comment]
> - Tokenizer version 4.4.4
> - I have tried it with all other modules disabled but 1 (The Ripper93's module hub, with I used for debugging)
> - I'm in the newest version DnD5e V5.0.2 which uses it's own character sheet system I think. But if that is what you mean then yes
> - I get one error when trying to open tokenizer this way:
> ```
> Uncaught TypeError: el.parentElement is null
>     _updatePosition foundry.mjs:27388
>     setPosition foundry.mjs:27364
>     _onSearchFilter foundry.mjs:29810
>     timeoutId foundry.mjs:692
>     setTimeout handler*debounce/< foundry.mjs:691
>     filter foundry.mjs:28660
>     bind foundry.mjs:28635
>     _onRender foundry.mjs:29527
>     _doEvent foundry.mjs:27654
>     #render foundry.mjs:27000
>     #try foundry.mjs:7130
>     add foundry.mjs:7103
>     add foundry.mjs:7101
>     render foundry.mjs:26925
>     render foundry.mjs:29407
>     browse foundry.mjs:29397
>     linkDefaultSheets hooks.js:342
>     jQuery 8
>     linkDefaultSheets hooks.js:320
>     jQuery 2
>     linkDefaultSheets hooks.js:315
>     #call foundry.mjs:23610
>     callAll foundry.mjs:23569
>     #callHooks foundry.mjs:27700
>     #dispatchEvent foundry.mjs:27684
>     _doEvent foundry.mjs:27657
>     promise callback*_doEvent foundry.mjs:27655
> foundry.mjs:27388:76
> ```
> 

I just installed  a brand new version of foundry, 5e and Tokenizer and it opens perfectly. Does this happen in a new world for you? Are you using an up-to-date version of chrome or the foundry app?

**Comment by Fabrice-Schoeneberger** [comment]
I just made a new world to test it and there it works perfectly ... very weird. The world I am running is very old, like 5 years. maybe there is something that starts to break things

**Comment by bmoser-pssolutions** [comment]
I have found that if in the tokenizer settings you delete the text in the text boxes for frames and masks, then save settings it will then work. Not sure what the problem is but this seems to solve it.

**Comment by MrPrimate** [comment]
> I have found that if in the tokenizer settings you delete the text in the text boxes for frames and masks, then save settings it will then work. Not sure what the problem is but this seems to solve it.

This is an unrelated issue, for the Application V2 character sheets that some systems are starting to use.

**Comment by MrPrimate** [comment]
@BokuNC @Veilza a sheet context option will now be added in v13.
